.github/workflows/run_pytest.yml

```diff
@@ -0,0 +1,106 @@
+name: Run pytest
+
+on:
+  pull_request:
+
+jobs:
+  install_and_test:
+
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash
+
+    container:
+      image: ghcr.io/dune-daq/nightly-release-alma9:development_v5
+
+    steps:
+      - name: Check URLs
+        run: |
+          echo https://github.com/${{ github.event.pull_request.head.repo.owner.login }}/druncschema.git
+          echo https://github.com/${{ github.event.pull_request.head.repo.owner.login }}/drunc.git
+
+      - name: Set up dev nightly
+        if: ${{ !cancelled() }}
+        run: |
+          mkdir nightly
+          cd nightly
+          source /cvmfs/dunedaq.opensciencegrid.org/setup_dunedaq.sh
+          setup_dbt latest_v5
+          dbt-create -n last_fddaq
+          cd NFD_DEV*
+          source env.sh
+          dbt-build
+          dbt-workarea-env
+
+      - name: Install druncschema and drunc
+        if: success()
+        run: |
+          cd nightly/NFD_DEV* || true
+          source env.sh
```

Comment:
Setting this environment based on the last nightly is used in a couple of repositories, and it is relatively complex. I would suggest to create a separate action that can be reused by any repo that needs it such that if changes are needed, they can be done in a single place.